src/uwtools/drivers/driver.py

```diff
@@ -1,177 +1,146 @@
 """
-Provides an abstract class representing drivers for various NWP tools.
+An abstract class for component drivers.
 """
-
-import os
-import shutil
+import re
 from abc import ABC, abstractmethod
-from collections.abc import Mapping
-from datetime import datetime
 from pathlib import Path
-from typing import Any, Dict, Type, Union
+from textwrap import dedent
+from typing import Any, Dict, List, Optional, Type
 
 from uwtools.config import validator
 from uwtools.config.formats.base import Config
 from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.exceptions import UWConfigError
 from uwtools.logging import log
-from uwtools.scheduler import BatchScript, JobScheduler
-from uwtools.types import DefinitePath, OptionalPath
+from uwtools.scheduler import JobScheduler
+from uwtools.types import DefinitePath
 
 
 class Driver(ABC):
     """
-    An abstract class representing drivers for various NWP tools.
+    An abstract class for component drivers.
     """
 
-    def __init__(
-        self,
-        config_file: DefinitePath,
-        dry_run: bool = False,
-        batch_script: OptionalPath = None,
-    ):
-        """
-        Initialize the driver.
+    def __init__(self, config_file: DefinitePath, dry_run: bool = False, batch: bool = False):
         """
+        A component driver.
 
-        self._config_file = config_file
-        self._dry_run = dry_run
-        self._batch_script = batch_script
+        :param config_file: Path to config file.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        self._config = YAMLConfig(config=config_file)
+        self._config.dereference()
         self._validate()
-        self._experiment_config = YAMLConfig(config=config_file)
-        self._platform_config = self._experiment_config.get("platform", {})
-        self._config: Dict[str, Any] = {}
-
-    # Public methods
+        self._dry_run = dry_run
+        self._batch = batch
 
-    @abstractmethod
-    def batch_script(self) -> BatchScript:
+    @staticmethod
+    def _create_user_updated_config(
+        config_class: Type[Config], config_values: dict, path: Path
+    ) -> None:
         """
-        Create a script for submission to the batch scheduler.
+        Create a config from a base file, user-provided values, of a combination of the two.
```

Comment:
I think you might have meant to say `Create a config from a base file, user-provided values, of a combination of the two.` --> `Create a config from a base file, user-provided values, *or* a combination of the two.`

Reply:
Yes, will fix, thanks!